libs/wingsdk/src/target-tf-azure/function.ts

```diff
@@ -182,27 +182,32 @@ export class Function extends cloud.Function {
       }) as any,
     });
 
-    // Add permissions to read function code
-    new RoleAssignment(this, `ReadLambdaCodeAssignment`, {
+    // Apply permissions from bound resources
+    for (const key of this.permissions?.keys() || []) {
```

Comment:
permissions is no longer optional, so I think we can remove
```suggestion
    for (const key of this.permissions.keys() || []) {
```